### ISSUE TITLE
[Bug]: Strange response from Retriever

### ISSUE THREAD

**Issue by c64er4ever** [run-llama/llama_index#8138]
### Bug Description

Hi
I am using Retriever to retrieve a list of nodes. For testing purposes I am always using the same query with similarity_top_k=10 and I am more or less getting the same results, with multiple nodes returned by the Retriever with high similarity scores. Today, at one time, I received back only a single node with Similarity of -0.010474686220271237. After repeating the same query, it returned the usual valid reply of multiple nodes with high similarity scores. Is there an explanation why this could happen, or is this a potential issue in the Retriever implementation?

Code snippet that uses the Retriever:
test_index = load_index_from_storage(storage_context, index_id=test_index_name)
retriever = test_index.as_retriever(similarity_top_k=10)
nodes = retriever.retrieve(query)
for node in nodes:
   print("Node ID(" + str(node.node.node_id) + ")")
   print("Similarity(" + str(node.score) + ")")

Thanks
Guy

### Version

0.8.36

### Steps to Reproduce

I could not reproduce this issue, it happened to me only once.

**Comment by logan-markewich** [comment]
If you are able to provide some reproducible steps/data, that would help narrow down the issue very quickly

**Comment by c64er4ever** [comment]
It may be worth mentioning that:
1. The LLM I'm using is Open AI
2. Immediately after this issue occurred, I noticed ServiceUnavailable exception from Open AI on another call. This issue was resolved after a few minutes.

Could this issue be related to Open AI not being available?  If yes, is there a way to get an indication from the Retriever that there was an error, so I can ignore the returned node and re-try?

Guy
